### ISSUE TITLE
Non-backwards compatible model movement in MSL3.2.1+build.3

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1757]
**Reported by beutlich on 4 Aug 2015 13:51 UTC**
84000601efc8b6900f1b42123930cb5ebe1baf74 moved Modelica.Media.Air.MoistAir.PsychrometricData to Modelica.Media.Examples.PsychrometricData. This breaks the backward compatibility of MSL3.2.1+build.3 as it actually requires a conversion script.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1757

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 4 Aug 2015 13:58 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 4 Aug 2015 14:00 UTC**
#1679 was the corresponding ticket.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 5 Aug 2015 06:28 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 24 Sep 2015 09:23 UTC**
This has been in fixed in trunk and the 3.2.1. maintenance branch in a8483d6f83e5682c08cf721b33650fc707a99517 and 288c056eded043b197bfd0d1752d78d09060c60c respectively, and also been marked with the obsolete icon for later removal

**Comment by modelica-trac-importer** [comment]
**Modified by hubertus on 24 Sep 2015 09:23 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 24 Sep 2015 22:06 UTC**
Since there actually was a fix after MSL 3.2.1+build.3 let's not reschedule the old ticket but rather create a new one: #1789.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 24 Sep 2015 22:06 UTC**
Add dummy example model Modelica.Media.Air.MoistAir.PsychrometricData to fix a non-backwards compatible model movement in MSL 3.2.1+build.3

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 24 Sep 2015 22:08 UTC**

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 24 Sep 2015 22:08 UTC**
Added a dummy example model Modelica.Media.Air.MoistAir.PsychrometricData to fix a non-backwards compatible model movement in MSL 3.2.1+build.3

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 Sep 2015 10:15 UTC**
There seems to be no way to fix this in a good way:

  PsychrometricData needs to be moved as stated in #1679 in order to remove a circular reference. If this is done, then the library is no longer backwards compatible (an example model was moved).

My conclusion is that the circular reference is much more critical and that this circular reference should be removed. This leads to a non-backwards compatible change, but this seems to be a minor (uncritical) issue because the model is an example model and it is very unlikely that this model is referenced in a user model. Therefore, I suggest to delete Modelica.Media.Air.MoistAir.PsychrometricData again (will do this in about 1 hour, if no one complains)

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 Sep 2015 10:49 UTC**
Well, it is a trade-off between a break in backwards compatibilty anc the cyclic reference. If we decide to go for breaking the backwards compatibilty it should be clearly mentioned in the release notes and #1789 should be closed simultaneously.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 Sep 2015 11:26 UTC**
Fixed in 0e51d82cc730312d85a9a1895ec9522f5b7f9ca8 for maintenance and in 410b89b65cef9849850447d464dd698d6886f5ec for trunk by 

- removing Modelica.Media.Air.MoistAir.PsychrometricData (since fixing #1679 is more important, and moving an example model is less critical because most likely not used by a model).

- documenting this in the release notes of build 4

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 Sep 2015 11:32 UTC**
The release notes of this ticket do no longer apply then.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 Sep 2015 11:36 UTC**
Sorry, do not understand you.
We cannot change build 3. In build 4, I have documented the issue of this ticket and of ticket #1679 in the release notes. Therefore, I close this ticket. If you do not agree, explain precisely what you think should be done to close this ticket.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 Sep 2015 11:40 UTC**
I added "Added a dummy example model Modelica.Media.Air.MoistAir.PsychrometricData to fix a non-backwards compatible model movement in MSL 3.2.1+build.3" to the release notes of this ticket. However you removed this dummy model again.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 Sep 2015 11:43 UTC**
Thanks for the clarification. Fixed the release notes of this ticket

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 28 Sep 2015 11:43 UTC**
Added an explanation in the build 4 release notes, why this (minor) non-backwards compatible change was made

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 Sep 2015 19:25 UTC**
Should a conversion script be provided anyway?

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 28 Sep 2015 21:01 UTC**
Replying to [comment:15 beutlich]:
> Should a conversion script be provided anyway?

I don't think so: I don't know a use case where that would make sense for an example. Is there a standard way of doing that now btw?

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 28 Sep 2015 21:22 UTC**
Replying to [comment:16 hubertus]:
> Is there a standard way of doing that now btw?

Conversion scripts are Dymola specific right now (see under Resources\Scripts\Dymola). But MCP14 (which clarifies the conversion scripts) could be integrated already to MLS 3.3r2.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 29 Sep 2015 11:07 UTC**
Replying to [comment:15 beutlich]:
> Should a conversion script be provided anyway?

I also don't think, because (a) it seems that only Dymola has conversion scripts (so no standardized solution), (b) handling of MSL with conversion script (also in Dymola) will become inconvenient/threatening for the user (users would then avoid to use build.4) and (c) it is an example model that was by accident in the wrong package and I am not aware that anyone is using example models as instances in his/her models.

To summarize: The current approach to document the situation in the release notes is the best approach.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 29 Sep 2015 11:11 UTC**
As a side remark, you see that I used

```
(a), (b), (c), (d)
```

in my input, and the third item, (c), was changed to the copyright symbol. Why, I do not know. This happens quite often to me that I write standard text (interpreted as some markdown dialect) and suddenly the rendered text does not look as I expect it.
